charon/src/bin/charon-driver/translate/translate_bodies.rs

```diff
@@ -890,13 +908,34 @@ impl BodyTransCtx<'_, '_, '_> {
                 let targets: Vec<(ScalarValue, BlockId)> = targets
                     .iter()
                     .map(|(v, tgt)| {
-                        let v = ScalarValue::from_le_bytes(int_ty, v.data_le_bytes);
+                        let v = ScalarValue::from_bytes(IntegerTy::Signed(int_ty), v.data_le_bytes);
```

Comment:
Let's keep the `from_le_bytes` name, a plain `from_bytes` doesn't say how the bytes are to be interpreted.

Reply:
Is the little endian encoding fixed by hax and rustc as well or is it the native endian encoding? I've changed the name and used functions such that hopefully Charon would also work on big endian targets by using the native endian encoding (not sure whether this would ever happen, but it's not too bad I guess).

Reply:
This is explicitly the output of `u128::to_le_bytes`. Tbf I'm not clear what the original `u128` was meant to encode; it's the `PU128` in [SwitchTargets](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/struct.SwitchTargets.html) if you're curious.